### ISSUE TITLE
Regression: using shinyAppDir() in shiny server doesn't load resources

### ISSUE THREAD

**Issue by daattali** [rstudio/shiny#2388]
I noticed that all my shiny apps on my shiny server that run using a call to `shinyAppDir()` rather than having their files locally are all of a sudden broken. The css and any other resources are not loaded correctly (resulting in a very broken looking app).

For example, making a test app at `/srv/shiny-server/test/app.R`:

```
dir <- system.file("examples", "demo", package = "shinyalert")
setwd(dir)
shiny::shinyAppDir(".")
```

(I have to setwd instead of just provide the path because of https://github.com/rstudio/shiny/issues/1063)

This works when I have shiny v1.2.0 installed on my shiny server, but breaks when I install 1.3.0 or 1.3.1 or the latest version as of right now

**Comment by jcheng5** [comment]
Thanks for the report. We'll try to repro tomorrow, but in the meantime you can try branch v1.3-patch.

**Comment by jcheng5** [comment]
OK, reproed the issue, and `v1.3-patch` does indeed fix it. So the upcoming Shiny 1.3.x release should take care of you.

We may want to consider a more specific fix for v1.4 though. `v1.3-patch` fixes it by restoring an old code path for serving the `www` directory out of R, instead of using the new httpuv static file serving codepaths that are faster (all `www` files except for `index.html` are supposed to come out of that codepath).

The reason these files aren't being served out of the static file codepaths is because Shiny Server launches `/srv/shiny-server/test/app.R` by calling `runApp()` on that file, which creates a shiny app object by sourcing app.R; but running app.R causes a shiny app object to be created by calling `shinyAppDir()`. The latter shiny app object has a `$staticPaths` element, but the former shiny app object doesn't pay attention to it. The fix is to make sure that any `$staticPaths` on inner shiny app objects, are respected by outer shiny app objects. (An open question is whether inner or outer should take precedence, if both have a www directory; we should just do whatever Shiny 1.2 does, probably.)

**Comment by jcheng5** [comment]
Repro without Shiny Server:

1. Create a file `app.R` with Dean's code above
2. Call `runApp("app.R")`

**Comment by cpsievert** [comment]
Closing since this appears to be fixed by 1.3.2, but please reopen if I'm missing something
